migration/v2.24/apply/01-fix-cp-node-taints.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+# Ensures control plane nodes have taint "node-role.kubernetes.io/control-plane:NoSchedule"
+# and removes old "node-role.kubernetes.io/master:NoSchedule" taint.
+
+HERE="$(dirname "$(readlink -f "${0}")")"
+ROOT="$(readlink -f "${HERE}/../../../")"
+
+CK8S_CLUSTER="${1}"
+export CK8S_CLUSTER
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+# functions currently available in the library:
+#   - logging:
+#     - log_info(_no_newline) <message>
+#     - log_warn(_no_newline) <message>
+#     - log_error(_no_newline) <message>
+#     - log_fatal <message> # this will call "exit 1"
+#
+#  - yq:
+#     - yq_null <sc|wc> <file> <target>
+#     - yq_copy <sc|wc> <file> <source> <destination>
+#     - yq_move <sc|wc> <file> <source> <destination>
+#     - yq_remove <sc|wc> <file> <target>
+#     - yq_length <sc|wc> <file> <target>
+
+
+function get_cp_nodes() {
+  #shellcheck disable=SC2207
+  CP_NODES=($(kubectl get no -l node-role.kubernetes.io/control-plane -oyaml | yq4 ".items[].metadata.name"))
+  export CP_NODES
+}
```

Comment:
Does it make more sense to have this function just printing the result so the other functions can insert them into local list?

```bash
function get_cp_nodes() {
  kubectl get no -l node-role.kubernetes.io/control-plane -oyaml | yq4 ".items[].metadata.name"
}

function add_control_plane_taint() {
  local cp_nodes=()
  mapfile -t cp_nodes < <(get_cp_nodes)
  
  for node in "${cp_nodes[@]}"; do
  .
  .
  .
```

It reduces the risk of it getting somewhere where it shouldn't be

Reply:
Sure, that also works